inc/saitam.h

```diff
@@ -1334,6 +1334,30 @@ typedef enum _sai_tam_report_attr_t
      */
     SAI_TAM_REPORT_ATTR_REPORT_INTERVAL_UNIT,
 
+    /**
+     * @brief Set ID for IPFIX template
+     *
+     * According to the IPFIX spec, the available range should be 256-65535.
+     * The value 0 means the ID will be decided by the vendor's SAI.
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @isvlan false
+     * @default 0
+     * @validonly SAI_TAM_REPORT_ATTR_TYPE == SAI_TAM_REPORT_TYPE_IPFIX
+     */
+    SAI_TAM_REPORT_ATTR_REPORT_IPFIX_TEMPLATE_ID,
+
+    /**
+     * @brief Query IPFIX template
+     *
+     * Return the IPFIX template binary buffer
+     *
+     * @type sai_u8_list_t
+     * @flags READ_ONLY
+     */
+    SAI_TAM_REPORT_ATTR_IPFIX_TEMPLATE,
```

Comment:
for sessions that exceeds the max length of template or data, we need to split the template into multiple templates:

- Modify the template type to sai_u8_list_list_t
- Rename SAI_TAM_REPORT_ATTR_IPFIX_TEMPLATE to SAI_TAM_REPORT_ATTR_IPFIX_TEMPLATE_LIST
- Rename SAI_TAM_REPORT_ATTR_REPORT_IPFIX_TEMPLATE_ID to SAI_TAM_REPORT_ATTR_REPORT_IPFIX_BASE_TEMPLATE_ID

Each template data has the template id in it, so we don't need another attribute or field to specify the template id list.